### PR TITLE
Spike fetch historical data from sql using cdc pipeline

### PR DESCRIPTION
**Pull request by HenrikSommer**
## Description

Add pre- and postdeployment scripts to pipeline and test new dbUp EnvironmentFilters.

### CODE REVIEW

**Comment by bemwen** [comment]
Since this PR is build on top of #82 it contains the same changes. This should clear up once we get #82 into main.

File: .github/actions/charges-apply-db-migrations/action.yml [hunk 1]
```diff
@@ -41,5 +41,5 @@ runs:
           echo "INCLUDE_TESTDATA=includeTestData" >> $GITHUB_ENV
         fi
       shell: bash    
-    - run: dotnet run --project source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/ --configuration Release -- "Server=tcp:${{ inputs.sql-server-url }},1433;Initial Catalog=${{ inputs.sql-database-name }};Persist Security Info=False;User ID=${{ env.CHARGES_USERNAME }};Password=${{ env.CHARGES_PASSWORD }};MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;" includeSeedData ${{ env.INCLUDE_TESTDATA }}
+    - run: dotnet run --project source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/ --configuration Release -- "Server=tcp:${{ inputs.sql-server-url }},1433;Initial Catalog=${{ inputs.sql-database-name }};Persist Security Info=False;User ID=${{ env.CHARGES_USERNAME }};Password=${{ env.CHARGES_PASSWORD }};MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;" includeSeedData ${{ env.INCLUDE_TESTDATA }} includePreDeploy includePostDeploy
```
Comment (BjarkeMeier) [inline .github/actions/charges-apply-db-migrations/action.yml:44]: Would anyone ever not want to execute pre- and post scripts? How about making them default?

**Review by Mech0z — APPROVED** [review]

**Review by BjarkeMeier — APPROVED** [review]
LGTM

File: .github/actions/charges-apply-db-migrations/action.yml (hunk 1 above)
Comment (HenrikSommer) [reply .github/actions/charges-apply-db-migrations/action.yml:44]: Could be relevant for testing purposes.

**Pull request merged**